analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -375,52 +376,59 @@ public void Finally_WrappedInLocalLifetimeRegion()
         public void Finally_ThrowInTry()
         {
             const string code = @"
-Tag(""BeforeTry"");
+var tag = ""BeforeTry"";
 try
 {
     Tag(""InTry"");
     throw new System.Exception();
-    Tag(""UnreachableInTry"");
+    tag = ""UnreachableInTry"";
 }
 finally
 {
-    Tag(""InFinally"");
+    tag = ""InFinally"";
 }
-Tag(""UnreachableAfterFinally"");";
-            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+tag = ""UnreachableAfterFinally"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
                 "BeforeTry",
                 "InTry",
-                "InFinally");
+                "InFinally",                // With Exception thrown by Tag("InTry")
+                "InFinally");               // With Exception thrown by `throw`
+
+            ValidateHasOnlyUnknownExceptionAndException(validator, "InFinally");
         }
 
         [TestMethod]
         public void Finally_ThrowInFinally()
         {
             const string code = @"
-Tag(""BeforeTry"");
+var tag = ""BeforeTry"";
 try
 {
-    Tag(""InTry"");
+    tag = ""InTry"";
 }
 finally
 {
-    Tag(""InFinally"");
+    tag = ""InFinally"";
     throw new System.Exception();
-    Tag(""UnreachableInFinally"");
+    tag  = ""UnreachableInFinally"";
 }
-Tag(""UnreachableAfterFinally"");";
-            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+tag = ""AfterFinally"";";
```

Comment:
The last update changed this from `UnreachableAfterFinally`, I'm not sure why. Let's keep the `UnreachableAfterFinally`
```suggestion
tag = ""UnreachableAfterFinally"";";
```